### ISSUE TITLE
Rails with Raygun gem but not setup throws own errors, hides app errors.

### ISSUE THREAD

**Issue by ylg** [MindscapeHQ/raygun4ruby#84]
We've this in an initializer:

```
if ENV['RAYGUN_APIKEY'].present?
  Raygun.setup do |config|
    config.api_key = ENV['RAYGUN_APIKEY']
    config.filter_parameters = Rails.application.config.filter_parameters
  end
end
```

When that doesn't run (because RAYGUN_APIKEY isn't there) then all application errors are replaced with Raygun's own error:

```
2016-04-07T16:18:17.858804+00:00 app[web.1]: Completed 500 Internal Server Error in 26ms
2016-04-07T16:18:17.859709+00:00 app[web.1]: [Raygun] Tracking Exception...
2016-04-07T16:18:17.859737+00:00 app[web.1]: [RAYGUN] Just a note, you've got no API Key configured, which means we can't report exceptions. Specify your Raygun API key using Raygun#setup (find yours at https://app.raygun.io)
2016-04-07T16:18:17.976198+00:00 app[web.1]: 
2016-04-07T16:18:17.976209+00:00 app[web.1]: NoMethodError (undefined method `strip' for nil:NilClass):
2016-04-07T16:18:17.976210+00:00 app[web.1]:   vendor/ruby-2.3.0/lib/ruby/2.3.0/net/http/header.rb:18:in `block in initialize_http_header'
2016-04-07T16:18:17.976211+00:00 app[web.1]:   vendor/ruby-2.3.0/lib/ruby/2.3.0/net/http/header.rb:16:in `each'
2016-04-07T16:18:17.976211+00:00 app[web.1]:   vendor/ruby-2.3.0/lib/ruby/2.3.0/net/http/header.rb:16:in `initialize_http_header'
2016-04-07T16:18:17.976212+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/httparty-0.13.7/lib/httparty/request.rb:176:in `setup_raw_request'
2016-04-07T16:18:17.976213+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/httparty-0.13.7/lib/httparty/request.rb:114:in `perform'
2016-04-07T16:18:17.976214+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/httparty-0.13.7/lib/httparty.rb:545:in `perform_request'
2016-04-07T16:18:17.976215+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/httparty-0.13.7/lib/httparty.rb:492:in `post'
2016-04-07T16:18:17.976215+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/raygun4ruby-1.1.9/lib/raygun/client.rb:163:in `create_entry'
2016-04-07T16:18:17.976216+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/raygun4ruby-1.1.9/lib/raygun/client.rb:27:in `track_exception'
2016-04-07T16:18:17.976217+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/raygun4ruby-1.1.9/lib/raygun.rb:43:in `track_exception'
2016-04-07T16:18:17.976218+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/raygun4ruby-1.1.9/lib/raygun/middleware/rack_exception_interceptor.rb:12:in `rescue in call'
2016-04-07T16:18:17.976218+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/raygun4ruby-1.1.9/lib/raygun/middleware/rack_exception_interceptor.rb:10:in `call'
2016-04-07T16:18:17.976219+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/newrelic_rpm-3.15.0.314/lib/new_relic/agent/instrumentation/middleware_tracing.rb:96:in `call'
2016-04-07T16:18:17.976220+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.6/lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
2016-04-07T16:18:17.976221+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/newrelic_rpm-3.15.0.314/lib/new_relic/agent/instrumentation/middleware_tracing.rb:96:in `call'
2016-04-07T16:18:17.976222+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.6/lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
2016-04-07T16:18:17.976222+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/newrelic_rpm-3.15.0.314/lib/new_relic/agent/instrumentation/middleware_tracing.rb:96:in `call'
2016-04-07T16:18:17.976223+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/railties-4.2.6/lib/rails/rack/logger.rb:38:in `call_app'
2016-04-07T16:18:17.976224+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/railties-4.2.6/lib/rails/rack/logger.rb:20:in `block in call'
2016-04-07T16:18:17.976225+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.6/lib/active_support/tagged_logging.rb:68:in `block in tagged'
2016-04-07T16:18:17.976225+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.6/lib/active_support/tagged_logging.rb:26:in `tagged'
2016-04-07T16:18:17.976226+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.6/lib/active_support/tagged_logging.rb:68:in `tagged'
2016-04-07T16:18:17.976227+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/railties-4.2.6/lib/rails/rack/logger.rb:20:in `call'
2016-04-07T16:18:17.976228+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/newrelic_rpm-3.15.0.314/lib/new_relic/agent/instrumentation/middleware_tracing.rb:96:in `call'
2016-04-07T16:18:17.976229+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.6/lib/action_dispatch/middleware/request_id.rb:21:in `call'
2016-04-07T16:18:17.976229+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/newrelic_rpm-3.15.0.314/lib/new_relic/agent/instrumentation/middleware_tracing.rb:96:in `call'
2016-04-07T16:18:17.976239+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/rack-1.6.4/lib/rack/methodoverride.rb:22:in `call'
2016-04-07T16:18:17.976240+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/newrelic_rpm-3.15.0.314/lib/new_relic/agent/instrumentation/middleware_tracing.rb:96:in `call'
2016-04-07T16:18:17.976241+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/rack-1.6.4/lib/rack/runtime.rb:18:in `call'
2016-04-07T16:18:17.976241+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/newrelic_rpm-3.15.0.314/lib/new_relic/agent/instrumentation/middleware_tracing.rb:96:in `call'
2016-04-07T16:18:17.976242+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/activesupport-4.2.6/lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
2016-04-07T16:18:17.976243+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/newrelic_rpm-3.15.0.314/lib/new_relic/agent/instrumentation/middleware_tracing.rb:96:in `call'
2016-04-07T16:18:17.976243+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.6/lib/action_dispatch/middleware/static.rb:120:in `call'
2016-04-07T16:18:17.976244+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/newrelic_rpm-3.15.0.314/lib/new_relic/agent/instrumentation/middleware_tracing.rb:96:in `call'
2016-04-07T16:18:17.976245+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/rack-1.6.4/lib/rack/sendfile.rb:113:in `call'
2016-04-07T16:18:17.976245+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/newrelic_rpm-3.15.0.314/lib/new_relic/agent/instrumentation/middleware_tracing.rb:96:in `call'
2016-04-07T16:18:17.976246+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/actionpack-4.2.6/lib/action_dispatch/middleware/ssl.rb:24:in `call'
2016-04-07T16:18:17.976247+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/newrelic_rpm-3.15.0.314/lib/new_relic/agent/instrumentation/middleware_tracing.rb:96:in `call'
2016-04-07T16:18:17.976253+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/railties-4.2.6/lib/rails/engine.rb:518:in `call'
2016-04-07T16:18:17.976253+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/railties-4.2.6/lib/rails/application.rb:165:in `call'
2016-04-07T16:18:17.976254+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/newrelic_rpm-3.15.0.314/lib/new_relic/agent/instrumentation/middleware_tracing.rb:96:in `call'
2016-04-07T16:18:17.976255+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/puma-3.1.0/lib/puma/configuration.rb:227:in `call'
2016-04-07T16:18:17.976255+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/puma-3.1.0/lib/puma/server.rb:561:in `handle_request'
2016-04-07T16:18:17.976256+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/puma-3.1.0/lib/puma/server.rb:406:in `process_client'
2016-04-07T16:18:17.976256+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/puma-3.1.0/lib/puma/server.rb:271:in `block in run'
2016-04-07T16:18:17.976257+00:00 app[web.1]:   vendor/bundle/ruby/2.3.0/gems/puma-3.1.0/lib/puma/thread_pool.rb:111:in `block in spawn_thread'
```

I would expect the behavior of Raygun to be not to intercept, handle, hide, whatever errors until #setup is called. Then, if we set it up wrong, sure, throw errors all over the place. :)

**Comment by nikz** [comment]
@ylg yeah, you're absolutely right... we'd need to change [this initialiser](https://github.com/MindscapeHQ/raygun4ruby/blob/master/lib/raygun/railtie.rb#L3) to run after all the others and check if Raygun is configured.

**Comment by nikz** [comment]
@ylg sorry for the slowness on this one, could you try out [this branch](https://github.com/MindscapeHQ/raygun4ruby/pull/87) for me and let me know how you get on?

**Comment by UberMouse** [comment]
Resolved by #87
